Tools/TestWebKitAPI/Tests/WebCore/IPAddressTests.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#include <WebCore/DNS.h>
+#include <wtf/text/WTFString.h>
+
+namespace TestWebKitAPI {
+
+#if OS(UNIX)
+
+TEST(IPAddressTests, MatchingNetMaskLength)
+{
+    auto address1 = WebCore::IPAddress::fromString("17.100.120.255"_s);
+    auto address2 = WebCore::IPAddress::fromString("17.100.100.255"_s);
+    auto address3 = WebCore::IPAddress::fromString("2001:db8::1234:5678"_s);
+    auto address4 = WebCore::IPAddress::fromString("2001:db8::1111:0000"_s);
+    auto address5 = WebCore::IPAddress::fromString("::1234:5678"_s);
+    auto address6 = WebCore::IPAddress::fromString("::"_s);
```

Comment:
Do we know that the IP addresses we get are always formed this way? Because the IP address spec allows form some weird formats which have historically been used to circumvent IP address detectors. Maybe that won't matter here because we're failing safe, i.e. failing *with* cap?

Reply:
That's a good question — from code inspection, it seems like CFNetwork calls into libnetcore, which then calls into `getnameinfo`, which ultimately uses `inet_ntop`. Since we're using the symmetric API (`inet_pton`) to parse this string back into an IP address, I believe we _should_ be safe here in terms of being able to parse the address strings that CFNetwork hands us.